### PR TITLE
build with latest docker api, fix go get issue

### PR DESCRIPTION
**Pull request by tonymet**
* Fixes build issue with rename of `Sirupsen/logrus` → `sirupsen/logrus`
* update docker api to v20
* update remaining deps with `go get -u all`
* improve call to `containerWait()` to wait on success & err
* replace `containerKill` with `containerStop`

## BEFORE
Go get was failing
```
go get .
go get: github.com/Sirupsen/logrus@none updating to
        github.com/Sirupsen/logrus@v1.8.1: parsing go.mod:
        module declares its path as: github.com/sirupsen/logrus
                but was required as: github.com/Sirupsen/logrus

```

## AFTER
```
go get .
# success!
```

## Testing
unit suite
```
$ go test .
ok      github.com/jdevelop/repull      0.063s
```
e2e
```
$ docker restart watchtower
$ pi@pi4plus:~/src/repull $ go run . -v watchtower
DEBU[0000] inspecting container 9c1e389a60bfb5550f6df1e52305ff43aeb3681e1d941b0560aed94444a97d1c 
DEBU[0000] pulling a new image for container 9c1e389a60bfb5550f6df1e52305ff43aeb3681e1d941b0560aed94444a97d1c : [/watchtower] 
DEBU[0000] pulling image containrrr/watchtower          
INFO[0000] using default tag: latest                    
INFO[0001] received image id: sha256:3283e0b5be326d77ff4f4e8b7a91d46aaa1d511c74877b5a32f161548812d00c 
DEBU[0001] stopping container 9c1e389a60bfb5550f6df1e52305ff43aeb3681e1d941b0560aed94444a97d1c : [/watchtower] 
DEBU[0002] removing container 9c1e389a60bfb5550f6df1e52305ff43aeb3681e1d941b0560aed94444a97d1c : [/watchtower] 
DEBU[0002] replacing image containrrr/watchtower with sha256:3283e0b5be326d77ff4f4e8b7a91d46aaa1d511c74877b5a32f161548812d00c 
INFO[0003] started container f54560e7cd1369e26049dbc5564244cad9603b68c7ea079da65d64ef0c1ae20b 
```

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -43,7 +44,8 @@ func main() {
 	if err != nil {
 		log.WithError(err).Fatalf("can't list containers")
 	}
-	ctx := context.Background()
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
```
Comment (jdevelop) [inline main.go:49]: Would it be possible to add a flag with the timeout, so a user can override one? Making it default to 5 seconds might not be sufficient for all environments.

**Review by jdevelop — CHANGES_REQUESTED** [review]
Awesome job, thanks! 

Just a small thing to make sure that the context timeout is configurable will do the trick.

:1st_place_medal:

File: main.go (hunk 1 above)
Comment (tonymet) [reply main.go:49]: good call i'll add a flag

File: main.go [hunk 2]
```diff
@@ -14,6 +15,7 @@ import (
 var (
 	log     = logrus.NewEntry(logrus.New())
 	verbose = flag.Bool("v", false, "verbose")
+	timeout = flag.Int("t", 5, "timeout")
```
Comment (jdevelop) [inline main.go]:
```suggestion
	timeout = flag.Int("t", 5, "timeout (seconds)")
```

**Review by jdevelop — APPROVED** [review]

**Comment by jdevelop** [comment]
Thanks a lot @tonymet !

**Comment by tonymet** [comment]
great teamwork thanks for the clear feedback and quick merge. 

It's a great tool – i've always wanted docker to include this in the CLI

**Comment by jdevelop** [comment]
I have some idea of adding another option to inspect the running container and print out the docker command line that was used to create one. I found a few tools that do this, but unfortunately they are either in Python or NodeJS, so getting all deps on the host just for this doesn't seem to be feasible.

**Pull request merged**